Automattic/sensei
# Add setup wizard data manager

### Changes proposed in this Pull Request

* Talking with @yscik, we decided to introduce a data manager to the setup wizard, using the `@wordpress/data`. It's a better approach because we're handling so much data and requests in the setup wizard and a data store will help us with some features.
* As a first case, I refactored the Welcome step data.
* The idea is to remove the useOnboardingApi in favor of the `@wordpress/data` store.

### Testing instructions

* Go to `/wp-admin/admin.php?page=sensei_onboarding&step=welcome`
* Click on "Continue"
* Toggle the select and click on "Continue"
* Go to `/wp-admin/admin.php?page=sensei-settings` and make sure the `Enable usage tracking` is updated.
